lib/sinatra/base.rb

```diff
@@ -1803,7 +1803,7 @@ class << self
     set :handler_name, nil
     set :traps, true
     set :server, %w[HTTP webrick]
-    set :bind, Proc.new { development? ? 'localhost' : '0.0.0.0' }
+    set :bind, Proc.new { development? ? 'localhost' : '127.0.0.1' }
```

Comment:
Uhm, this makes `127.0.0.1` the default for everything not `development`, I don't think that is something anyone want to change?

#1369 ("Change default host to 0.0.0.0 in development mode") that you refer to is about going back to `0.0.0.0` in `development` (i.e. reverting the change from 2013: https://github.com/sinatra/sinatra/commit/0f9a959253bdf0c39422ffb9b60278df46ef0248, https://github.com/sinatra/sinatra/pull/634)